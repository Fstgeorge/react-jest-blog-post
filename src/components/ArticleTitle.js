import React, { PropTypes } from 'react';

const ArticleTitle = ({ title, url, timestamp }) => (
  <div>
    <h1>{title} [{timestamp}]</h1>
    <h4><a href={url}>{url}</a></h4>
  </div>
);

ArticleTitle.propTypes = {
  title: PropTypes.string.isRequired,
  timestamp: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

export default ArticleTitle;
