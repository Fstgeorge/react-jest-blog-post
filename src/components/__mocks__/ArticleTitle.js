import React from 'react';

const ArticleTitle = props => (
  <span>ArticleTitle {JSON.stringify(props)}</span>
);

module.exports = ArticleTitle;
