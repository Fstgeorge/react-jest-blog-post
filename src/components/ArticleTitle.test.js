import renderer from 'react-test-renderer';
import React from 'react';
import ArticleTitle from './ArticleTitle';

test('ArticleTest renders correctly', () => {
  const tree = renderer.create(
    <ArticleTitle title="A title" url="http://example.com" timestamp="2015-12-31" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
