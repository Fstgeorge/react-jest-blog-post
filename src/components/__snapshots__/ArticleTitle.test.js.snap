exports[`test ArticleTest renders correctly 1`] = `
<div>
  <h1>
    A title
     [
    2015-12-31
    ]
  </h1>
  <h4>
    <a
      href="http://example.com">
      http://example.com
    </a>
  </h4>
</div>
`;
